textproto/header.go

```diff
@@ -425,14 +425,34 @@ const (
 	maxLineOctets  = 4000
 )
 
+// ReadOptions represents the options in a header reader.
+type ReadOptions struct {
+	MaxHeaderLineLength int
+}
+
+func getDefaultReaderOptions(o *ReadOptions) *ReadOptions {
+	if o == nil {
+		o = &ReadOptions{}
+	}
+	if o.MaxHeaderLineLength == 0 {
+		o.MaxHeaderLineLength = maxLineOctets
+	}
+	return o
+}
+
 // ReadHeader reads a MIME header from r. The header is a sequence of possibly
 // continued Key: Value lines ending in a blank line.
-func ReadHeader(r *bufio.Reader) (Header, error) {
+//
+// Default value for MaxHeaderLineLength option
```

Comment:
This paragraph should probably be moved in the `ReadOptions` docs.

Would be nice to document that the parameter can be nil here.